packages/priority-queue/src/index.js

```diff
@@ -0,0 +1,50 @@
+const requestFrame = window.requestIdleCallback ? window.requestIdleCallback : window.requestAnimationFrame;
+
+export const createQueue = () => {
+	const waitingList = [];
+	const elementsMap = new WeakMap();
```

Comment:
Is there a reason for this to be a `WeakMap` ? Would we ever be taking advantage of the garbage collection behavior?

Reply:
We do use the garbage collection behavior. Imagine a component getting unmounted while it has an element in the queue. It gets removed.